mesonbuild/wrap/wraptool.py

```diff
@@ -90,30 +90,52 @@ def search(options: 'argparse.Namespace') -> None:
                 if dep.find(name) != -1:
                     print(f'Dependency {dep} found in wrap {p}')
 
-def get_latest_version(name: str, allow_insecure: bool) -> T.Tuple[str, str]:
-    releases = get_releases(allow_insecure)
+def find_latest_version(releases: T.Dict[str, T.Any], name: str) -> T.Tuple[str, str]:
     info = releases.get(name)
     if not info:
         raise WrapException(f'Wrap {name} not found in wrapdb')
     latest_version = info['versions'][0]
     version, revision = latest_version.rsplit('-', 1)
     return version, revision
 
-def install(options: 'argparse.Namespace') -> None:
-    name = options.name
-    if not os.path.isdir('subprojects'):
-        raise SystemExit('Subprojects dir not found. Run this script in your source root directory.')
+def get_latest_version(name: str, allow_insecure: bool) -> T.Tuple[str, str]:
+    return find_latest_version(get_releases(allow_insecure), name)
+
+def install_one(get_releases: T.Callable[[], T.Dict[str, T.Any]], name: str, allow_insecure: bool) -> None:
     if os.path.isdir(os.path.join('subprojects', name)):
-        raise SystemExit('Subproject directory for this project already exists.')
+        raise WrapException(f'Subproject directory for {name} already exists.')
     wrapfile = os.path.join('subprojects', name + '.wrap')
     if os.path.exists(wrapfile):
-        raise SystemExit('Wrap file already exists.')
-    (version, revision) = get_latest_version(name, options.allow_insecure)
-    url = open_wrapdburl(f'https://wrapdb.mesonbuild.com/v2/{name}_{version}-{revision}/{name}.wrap', options.allow_insecure, True)
+        raise WrapException(f'Wrap file for {name} already exists.')
+    (version, revision) = find_latest_version(get_releases(), name)
```

Comment:
Didn't you want to optionally get version from the name, something like:
```
wanted = name.rsplit('=', maxsplit=1)
if len(wanted) > 1:
  version, revision = wanted.rsplit('-', maxsplit=1)
else:
  version, revision = find_latest_version(get_releases(), name)
```

Reply:
Ideally it could search for latest that match the wanted version. Something like `foo=2` would find foo `2.1-1`.

Reply:
> I would like to address that in a separate PR.

Fair enough.